### ISSUE TITLE
Pygame 2.0.0 puts window at (0,0) when toggling back from full-screen, hiding the top bar. (worked fine in 1.9.6)

### ISSUE THREAD

**Issue by QuetzalQatl** [pygame/pygame#2360]
**Environment:**

- **Operating system** Windows 7
- **Python version** 3.7.9
- **SDL version** 2.0.12
- **PyGame version** 1.9.6 & 2.0.0 

**Current behavior:**

When toggling back from fullscreen, the window gets placed at coördinates (0,0), the uttermost top left of the screen. This hides the top bar, making it hard/impossible to grab the window. 

**Expected behavior:**

The 1.9.6 behavior: when toggling between fullscreen and non-fullscreen, the position of the non-fullscreen stays the same. If that is impossible for some reason: at least display the new non-fullscreen window a topbar height below the top, so users can adjust the position themselves. 

**Screenshots**

Before Fullscreen toggle 1.9.6:
![before1 9 6](https://user-images.githubusercontent.com/6069410/100325073-68042980-2fc8-11eb-8d4f-b0fd535d6323.png)
After Fullscreen toggle 1.9.6:
![after1 9 6](https://user-images.githubusercontent.com/6069410/100325083-6cc8dd80-2fc8-11eb-8c8a-89e760f7f7d0.png)
Before Fullscreen toggle 2.0.0:
![before2 0 0](https://user-images.githubusercontent.com/6069410/100324810-05ab2900-2fc8-11eb-8d1b-7a3f39757134.png)
After Fullscreen toggle 2.0.0:
![after2 0 0](https://user-images.githubusercontent.com/6069410/100324814-06dc5600-2fc8-11eb-9eda-9e8210eab783.png)

**Steps to reproduce:**

pip install pygame==1.9.6
run example code, press F twice
pip install pygame==2.0.0
run example code, press F twice

**Test code**

```
import sys
import pygame
from pygame.locals import *
 
class PygameProblem():
    def __init__(self):
        pygame.init()
        
        pgdi=pygame.display.Info()
        self.fullScreenSize=(pgdi.current_w, pgdi.current_h)
        self.nonFullScreenSize=(int(pgdi.current_w/2.0),int(pgdi.current_h/2.0))
        self.isFullScreen=False
        
        self.setDisplay()
        pygame.display.update()
        
        self.mainLoop()        
        
    def setDisplay(self):
        if self.isFullScreen:
            self.windowSurface = pygame.display.set_mode(self.fullScreenSize, pygame.FULLSCREEN ,0)
        else:
            self.windowSurface = pygame.display.set_mode(self.nonFullScreenSize, pygame.RESIZABLE ,0)
        
    def mainLoop(self):
        while True:
            for event in pygame.event.get():
                if event.type == QUIT:
                    pygame.quit()
                    sys.exit()
                elif event.type== KEYDOWN:
                    if event.key== K_f: # F toggles fullscreen
                        self.isFullScreen=not self.isFullScreen
                        self.setDisplay()
                        pygame.display.update()
if __name__=='__main__':
    PygameProblem()
```

**Comment by robertpfeiffer** [comment]
Should I try to work around this or report it with upstream? This doesn't happen on X11, and I assume it's a platform-dependent. It might not even be X11/windows-specific, but down to the window manager I'm using.

**Comment by MyreMylar** [comment]
Somebody also mentioned on a pygame discord that the `SDL_VIDEO_WINDOW_POS` wasn't working in pygame 2, though I haven't experimented myself yet - might have a tinker...

**Comment by MyreMylar** [comment]
Looks like `SDL_VIDEO_WINDOW_POS` still works. It sets the initial window position before the first call to `set_mode()` and thereafter it can't be changed with that environment variable.

**Comment by QuetzalQatl** [comment]
Both os.environ['SDL_VIDEO_CENTERED'] = '1'
and os.environ['SDL_VIDEO_WINDOW_POS'] = str(x) + "," + str(y) work fine: but only if you pygame.quit then set SDL, then pygame.init again, which seems a bit crude? 

Workaround code: 

```
import sys
import pygame
import os
from pygame.locals import *

class PygameProblem():
    def __init__(self):
        os.environ['SDL_VIDEO_WINDOW_POS'] = str(100)+ "," + str(100)
        #os.environ['SDL_VIDEO_CENTERED'] = '1'
        pygame.init()
        
        pgdi=pygame.display.Info()
        self.fullScreenSize=(pgdi.current_w, pgdi.current_h)
        self.nonFullScreenSize=(int(pgdi.current_w/2.0),int(pgdi.current_h/2.0))
        self.isFullScreen=False
        
        self.setDisplay()
        pygame.display.update()
        
        self.mainLoop()        
        
    def setDisplay(self):
        if self.isFullScreen:
            self.windowSurface = pygame.display.set_mode(self.fullScreenSize, pygame.FULLSCREEN ,0)
        else:
            pygame.quit() 
            os.environ['SDL_VIDEO_WINDOW_POS'] = str(100)+ "," + str(100)
            #os.environ['SDL_VIDEO_CENTERED'] = '1'
            pygame.init()
            self.windowSurface = pygame.display.set_mode(self.nonFullScreenSize, pygame.RESIZABLE ,0)
        
    def mainLoop(self):
        while True:
            for event in pygame.event.get():
                if event.type == QUIT:
                    pygame.quit()
                    sys.exit()
                elif event.type== KEYDOWN:
                    mods=pygame.key.get_mods()
                    if event.key== K_f: # F toggles fullscreen
                        self.isFullScreen=not self.isFullScreen
                        self.setDisplay()
                        pygame.display.update()
                        
if __name__=='__main__':
    PygameProblem()
```

**Comment by QuetzalQatl** [comment]
Both os.environ['SDL_VIDEO_CENTERED'] = '1'
and os.environ['SDL_VIDEO_WINDOW_POS'] = str(x) + "," + str(y) work fine: but only if you pygame.quit then set SDL, then pygame.init again, which seems a bit crude?

Workaround code:

```
import sys
import pygame
import os
from pygame.locals import *

class PygameProblem():
    def __init__(self):
        os.environ['SDL_VIDEO_WINDOW_POS'] = str(100)+ "," + str(100)
        #os.environ['SDL_VIDEO_CENTERED'] = '1'
        pygame.init()
        
        pgdi=pygame.display.Info()
        self.fullScreenSize=(pgdi.current_w, pgdi.current_h)
        self.nonFullScreenSize=(int(pgdi.current_w/2.0),int(pgdi.current_h/2.0))
        self.isFullScreen=False
        
        self.setDisplay()
        pygame.display.update()
        
        self.mainLoop()        
        
    def setDisplay(self):
        if self.isFullScreen:
            self.windowSurface = pygame.display.set_mode(self.fullScreenSize, pygame.FULLSCREEN ,0)
        else:
            pygame.quit() 
            os.environ['SDL_VIDEO_WINDOW_POS'] = str(100)+ "," + str(100)
            #os.environ['SDL_VIDEO_CENTERED'] = '1'
            pygame.init()
            self.windowSurface = pygame.display.set_mode(self.nonFullScreenSize, pygame.RESIZABLE ,0)
        
    def mainLoop(self):
        while True:
            for event in pygame.event.get():
                if event.type == QUIT:
                    pygame.quit()
                    sys.exit()
                elif event.type== KEYDOWN:
                    mods=pygame.key.get_mods()
                    if event.key== K_f: # F toggles fullscreen
                        self.isFullScreen=not self.isFullScreen
                        self.setDisplay()
                        pygame.display.update()
                        
if __name__=='__main__':
    PygameProblem()
    ```

**Comment by mcpalmer1980** [comment]
This bug occurs for me on Ubuntu 20.04. An easy solution is to not add the fullscreen flag at all, but to toggle fullscreen after creating the window first.

This works perfectly for me in Flyboy.

**Comment by QuetzalQatl** [comment]
This code works for me pretty flawlessly, only problem is toggling from a maximized window to fullscreen (bug?)
```
#!/usr/bin/env python3
import os
import pygame

pygame.init()

# find out & store desktopSize
pgdi=pygame.display.Info()
desktopSize=(pgdi.current_w, pgdi.current_h)

FPS = 100
clock = pygame.time.Clock()

screenWidth=160 # initial screen width when windowed
screenHeight=120

hor=0 # ball position on screen 
ver=0

# will make the first intance of the window appear in the middle of the screen.
# (consecutive instances will appear in the upper most left corner)
os.environ['SDL_VIDEO_CENTERED'] = '1' 

# flag to keep track of the question: is the window maximized or not
# should remain False at startup
maximized=False 

# choose to startup fullscreen or not.
# both options will work fine
isFullScreen=True 

# set up initial display:

# always make the non-fullscreen screen first, 
# so the fullscreen has something to toggle back to that is placed properly (as in not at 0,0 )
screen = pygame.display.set_mode((screenWidth,screenHeight), pygame.RESIZABLE) 

if isFullScreen:
    # set size to desktop 
    # not doing this takes much longer (about 4 seconds)
    # but doing this is near instantanious
    pygame.display.set_mode(desktopSize, pygame.RESIZABLE) 
    
    pygame.display.toggle_fullscreen() # now toggle fullscreen


done=False
while not done:
    for event in pygame.event.get():        
        if event.type == pygame.QUIT: 
            # like ctrl-break or the cross in the title bar
            done = True
        elif event.type == pygame.KEYDOWN and event.key == pygame.K_q: 
            # pressing q will also quit the program
            done = True
        elif event.type == pygame.KEYDOWN and event.key == pygame.K_f:
            # pressing f will toggle fullscreen
            
            isFullScreen=not isFullScreen # toggle fullscreen flag
            if isFullScreen: #toggling from not fullscreen to fullscreen
                if maximized: # abort 
                    # toggling from maximized to fullscreen gives a warning:
                    # Warning: re-creating window in toggle_fullscreen      
                    # then fail to function properly, giving back a large window that cannot be resized. 
                    # so, we abort this (till we know a better solution)
                    print('cannot toggle from maximized to fullscreen?')
                    isFullScreen=not isFullScreen # restore fullscreen flag
                else:
                    # store size of windowed screen 
                    # so we can reset it when toggling back
                    screenWidth=screen.get_width()
                    screenHeight=screen.get_height()
                
                    # set the number of pixels to desktop size
                    pygame.display.set_mode(desktopSize, pygame.RESIZABLE) 
                    # then toggle that to fullscreen                
                    pygame.display.toggle_fullscreen()                    
                    
            else: # toggling from fullscreen to not fullscreen
                pygame.display.toggle_fullscreen() 
                # restore to old size
                pygame.display.set_mode((screenWidth, screenHeight), pygame.RESIZABLE)
            
        # this is done to keep track of the question: 'is the window maximized or not':
        elif event.type ==pygame.WINDOWEVENT:
            if event.event == pygame.WINDOWEVENT_MAXIMIZED:
                maximized=True 
            elif event.event == pygame.WINDOWEVENT_MINIMIZED:
                maximized=False
            elif event.event == pygame.WINDOWEVENT_RESTORED:
                maximized=False
    
    # show some things on screen:        
    
    # paint background
    screen.fill((255, 0, 255))

    # slide ball position
    hor += 1
    hor = hor % screen.get_width()
    ver += hor % 2
    ver = ver % screen.get_height()
    
    # paint ball
    pygame.draw.circle(screen, (255, 255, 255), (hor, ver), 5)
        
    pygame.display.flip()
    clock.tick(FPS)
```

**Comment by eebop** [comment]
On a mac, it goes up to the top left hand corner, but the top bar is still visible.

**Comment by robertpfeiffer** [comment]
On ubuntu 20.10, I can reproduce this behaviour, but I could not at the time this bug was filed...

**Comment by Starbuck5** [comment]
I'm removing the windows label because this seems to affect all platforms. (Windows, Mac, and Ubuntu at least).

**Comment by illume** [comment]
Closed thanks to Starbuck5 in #2460

**Comment by CodeGuru12** [comment]
This does not appear to be fixed in pygame 2.0.1, using SDL 2.0.14, python 3.9.4 with windows 10. I am having this issue with the latest pygame release. Didn't realize this was bug with pygame.

**Comment by Starbuck5** [comment]
@CodeGuru12 The fix was merged in after 2.0.1, there just hasn't been a new release in a while. It will be fixed in 2.0.2.

In the meantime, you could use the workaround from QuetzalQatl above, or something like this: (Just be advised this will break in the future, as it uses experimental features).
```
import pygame
from pygame._sdl2 import Window, WINDOWPOS_CENTERED

screen = pygame.display.set_mode((1280,720))
window = Window.from_display_module()
fullscreen = False

clock = pygame.time.Clock()

while True:
    screen.fill("blue")
    
    for event in pygame.event.get():
        if event.type == pygame.KEYDOWN and event.key == pygame.K_f:
            fullscreen = not fullscreen
            pygame.display.set_mode(screen.get_size(), pygame.FULLSCREEN if fullscreen else 0)
            window.position = WINDOWPOS_CENTERED
        
        if event.type == pygame.QUIT:
            pygame.quit()
            raise SystemExit

    pygame.display.flip()
    clock.tick(144)
```
